pyquil/api/qpu.py

```diff
@@ -40,6 +41,22 @@ def get_devices(async_endpoint='https://job.rigetti.com/beta', api_key=None, use
     return {Device(name, device) for (name, device) in response.json()['devices'].items()}
 
 
+def append_measures_to_program(gate_program, qubits):
+    """
+    For run_and_measure programs, append MEASURE instructions to the
+    program, on all provided qubits.
+
+    :param Program gate_program: Program without MEASURE instructions
+    :param list qubits: Qubits to measure
+    :return: Full pyquil program with MEASUREs
+    :rtype: Program
+    """
+    meas_program = Program()
+    for q in qubits:
+        meas_program.inst(MEASURE(q, q))
+    return gate_program + meas_program
```

Comment:
Does this destructively modify the left operand? If so, users probably aren't going to expect their Program object to accrue a bunch of measures whenever they pass it to QPUConnection.run_and_measure().

Reply:
good call out -- this returns a new program so it is ok. i originally gate_program.inst(stuff) and that accumulation (fortunately) got caught by the UTs